mirrormanager2/checkin.py

```diff
@@ -40,6 +41,34 @@
 logger = logging.getLogger(__name__)
 
 
+@APP.route('/checkin/json', methods=['POST'])
+def checkin_json():
+    ctype = flask.request.headers.get('Content-Type')
+    try:
+        if ctype == 'application/gzip':
+            data = json.loads(zlib.decompress(flask.request.data, 16+zlib.MAX_WBITS))
+        elif ctype == 'application/json':
+            data = json.loads(flask.request.data)
+        else:
+            return 'Invalid content type', 415
+    except Exception as ex:
+        logging.error('Unable to read input data: %s' % ex)
+        return 'Error decoding', 400
```

Comment:
What Exceptions do you expect? It's often a nice pattern to handle specific expected Exceptions and then have a final catch-all like this, so you can provide nice error messages for the expected ones.